FairyTaleDefender/Assets/_Game/Scripts/Tests/Runtime/Systems/SaveGameSystem/ScriptableObjects/SaveGameManagerSOTests.cs

```diff
@@ -0,0 +1,189 @@
+				}
+
+				Directory.Move(SaveGamesBackupPath, SaveGamesPath);
+			}
+		}
+
+		private async UniTask<SaveGameMeta> PrepareSaveGameAsync(bool isValid = true, string name = "1")
+		{
+			var fileManager = new JsonFileManager();
+			name = name.Slugify();
+
+			var saveGamePath = Path.Combine(SaveGamesPath, name);
+			var metaPath = Path.Combine(saveGamePath, Constants.SaveGames.MetaFileName);
+
+			var meta = new SaveGameMeta()
+			{
+				Name = $"Unit Test Save {name}",
+				LastPlayedDate = DateTime.Now.Subtract(new TimeSpan(1)),
+				Directory = saveGamePath,
+			};
+
+			await fileManager.WriteAsync(metaPath, meta);
+
+			if (!isValid)
+			{
+				return meta;
+			}
+
+			var saveGameFilePath = Path.Combine(saveGamePath, Constants.SaveGames.SaveGameFileName);
+			await fileManager.WriteAsync(saveGameFilePath, new SaveGameData()
+			{
+				LastLevel = null
+			});
+
+			meta.Hash = SaveGameHash.Create(meta.Name, saveGameFilePath);
+			await fileManager.WriteAsync(metaPath, meta);
+
+			return meta;
+		}
+
+		[UnityTest]
+		public IEnumerator LoadAvailableSaveGamesAsync_CanListAvailableValidSaveGames() => UniTask.ToCoroutine(async () =>
+		{
+			await PrepareSaveGameAsync();
+
+			var sut = ScriptableObject.CreateInstance<SaveGameManagerSO>();
+
+			var metas = await sut.ListAvailableSaveGamesAsync();
+
+			metas.Count.Should().Be(1);
+			metas[0].Name.Should().Be("Unit Test Save 1");
+		});
+
+		[UnityTest]
+		public IEnumerator LoadAvailableSaveGamesAsync_CanListAvailableValidSaveGames_SkippingInvalidSaveGames() =>
+			UniTask.ToCoroutine(async () =>
+			{
+				await PrepareSaveGameAsync(isValid: false);
+				await PrepareSaveGameAsync(name: "2");
+
+				var sut = ScriptableObject.CreateInstance<SaveGameManagerSO>();
+
+				var metas = await sut.ListAvailableSaveGamesAsync();
+
+				metas.Count.Should().Be(1);
+				metas[0].Name.Should().Be("Unit Test Save 2");
+			});
+
+		[UnityTest]
+		public IEnumerator LoadSaveGameAsync_ReturnsNull_WhenSaveGameIsInvalid() =>
+			UniTask.ToCoroutine(async () =>
+			{
+				var meta = await PrepareSaveGameAsync(isValid: false);
+
+				var sut = ScriptableObject.CreateInstance<SaveGameManagerSO>();
+
+				var saveGame = await sut.LoadSaveGameAsync(meta);
+
+				saveGame.Should().BeNull();
+			});
+
+		[UnityTest]
+		public IEnumerator LoadSaveGameAsync_ReturnsASaveGame() =>
+			UniTask.ToCoroutine(async () =>
+			{
+				var meta = await PrepareSaveGameAsync();
+
+				var sut = ScriptableObject.CreateInstance<SaveGameManagerSO>();
+
+				var saveGame = (await sut.LoadSaveGameAsync(meta))!;
+
+				saveGame.Should().NotBeNull();
+				saveGame.Meta.Should().NotBeNull();
+				saveGame.Data.Should().NotBeNull();
+			});
+
+		[UnityTest]
+		public IEnumerator CreateSaveGameAsync_CreatesASaveGame() =>
+			UniTask.ToCoroutine(async () =>
+			{
+				var sut = ScriptableObject.CreateInstance<SaveGameManagerSO>();
+
+				var meta = await sut.CreateSaveGameAsync("UnitTest", new());
```

Comment:
Hier kommt das komplette SaveGame zurück, nicht nur die Meta.
```suggestion
				var saveGame = await sut.CreateSaveGameAsync("UnitTest", new());
```